anap1003/assignment3/Assignment3.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package assignment3;
+
+import java.util.BitSet;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+/**
+ *
+ */
+public class Assignment3 {
+    private int rows, cols;
+    private Set<String> foundWords;
+    private char[][] grid;
+    
+    /**
+     * 
+     * @param gridToSearch
+     * @param dictionary
+     * @return
+     * @throws Exception 
+     */
+    public Set wordSearch(char[][] gridToSearch, Dictionary dictionary) throws Exception {
+        foundWords = new LinkedHashSet();
+        
+        grid = gridToSearch;
+        rows = grid.length;
+        cols = grid[0].length;
+        BitSet visited = new BitSet(grid.length * grid[0].length);

```

Comment:
since you're setting rows and cols. I'd use them to initialize BitSet instead of calling .length again.